pkg/providers/tinkerbell/assert.go

```diff
@@ -313,6 +334,26 @@ func (v *ValidatableTinkerbellCAPI) WorkerNodeHardwareGroups() []WorkerNodeHardw
 	return workerNodeHardwareList
 }
 
+// ClusterK8sVersion returns the Kubernetes version in major.minor format for a ValidatableTinkerbellCAPI.
+func (v *ValidatableTinkerbellCAPI) ClusterK8sVersion() *v1alpha1.KubernetesVersion {
+	return v.extractK8sVersion(v.KubeadmControlPlane.Spec.Version)
+}
+
+// WorkerNodeGroupK8sVersion returns each worker node group mapped to Kubernetes version in major.minor format for a ValidatableTinkerbellCAPI.
+func (v *ValidatableTinkerbellCAPI) WorkerNodeGroupK8sVersion() map[string]*v1alpha1.KubernetesVersion {
+	WNGK8sversion := make(map[string]*v1alpha1.KubernetesVersion)
+	for _, wng := range v.WorkerGroups {
+		k8sVersion := v.extractK8sVersion(*wng.MachineDeployment.Spec.Template.Spec.Version)
+		WNGK8sversion[wng.MachineDeployment.Name] = k8sVersion
+	}
+	return WNGK8sversion
+}
+
+func (v *ValidatableTinkerbellCAPI) extractK8sVersion(k8sversion string) *v1alpha1.KubernetesVersion {
```

Comment:
What does this do? What's the expected format of `k8sVersion`?

Reply:
it is expected to be in the format we specify in our spec which is major.minor i.e. For ex. '1.xx'